src/internal/collection/postgres_collection.go

```diff
@@ -675,14 +675,15 @@ func (c *postgresReadOnlyCollection) WatchByIndexF(index *Index, indexVal string
 
 type postgresReadWriteCollection struct {
 	*postgresCollection
-	tx *pachsql.Tx
+	ctx context.Context
+	tx  *pachsql.Tx
 }
 
 func (c *postgresReadWriteCollection) Get(key interface{}, val proto.Message) error {
 	var result *model
 	var err error
 	err = c.withKey(key, func(rawKey string) error {
-		result, err = c.get(context.Background(), rawKey, c.tx)
+		result, err = c.get(c.ctx, rawKey, c.tx)
```

Comment:
Here, and throughout, the method should have the signature `Get(ctx context.Context, key any, val proto.Message) error` and the call here should be `c.get(ctx, rawKey, c.tx)`.